### PR TITLE
fix: [IDP-682]: Fix for integration config for proxy support and refactoring to process config at processor

### PR DESCRIPTION
**Pull request by devesh-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49339)

### CODE REVIEW

**Comment by devesh-harness** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger uts

File: idp-service/src/main/java/io/harness/idp/proxy/delegate/BUILD.bazel [hunk 1]
```diff
@@ -12,18 +12,10 @@ java_library(
...
-        "//clients/access-control/sdk:module",
         "//idp-service/contracts/openapi/v1:idp-service-server-spec-module",
         "//idp-service/contracts/src/main/java/io/harness/idp/annotations:module",
         "//idp-service/modules/clients:module",
-        "//idp-service/src/main/java/io/harness/idp/common:module",
         "//idp-service/src/main/java/io/harness/idp/common/delegateselectors/cache:module",
-        "//idp-service/src/main/java/io/harness/idp/envvariable:module",
-        "@maven//:com_google_inject_guice",
-        "@maven//:javax_javaee_api",
+        "//idp-service/src/main/java/io/harness/idp/proxy/ngmanager:module",
```
Comment (vikyathharekal) [inline idp-service/src/main/java/io/harness/idp/proxy/delegate/BUILD.bazel:19]: nit: Looks like we are using io.harness.idp.proxy.ngmanager.IdpAuthInterceptor.AUTHORIZATION. We can move this to common constants. I'll take it up in my next PR

File: idp-service/src/main/java/io/harness/idp/proxy/envvariable/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+load("//:tools/bazel/harness.bzl", "java_library")
+
+java_library(
+    name = "module",
+    srcs = glob(["**/*.java"]),
+    resources = [],
+    visibility = ["//idp-service:__subpackages__"],
+    deps = [
+        "//:lombok",
+        "//idp-service/src/main/java/io/harness/idp/common/delegateselectors/cache:module",
```
Comment (vikyathharekal) [inline idp-service/src/main/java/io/harness/idp/proxy/envvariable/BUILD.bazel:15]: we might not need dependency on delegateselectors here.

File: idp-service/src/main/java/io/harness/idp/gitintegration/processor/impl/GithubConnectorProcessor.java [hunk 3]
```diff
@@ -135,4 +150,9 @@ public String getLocationTarget(CatalogConnectorInfo catalogConnectorInfo, Strin
     return catalogConnectorInfo.getRepo() + SLASH_DELIMITER + SOURCE_FORMAT + SLASH_DELIMITER
         + catalogConnectorInfo.getBranch() + path;
   }
+
+  private String getGithubApiBaseUrlFromHost(String host) {
+    return (host.equals("github.com")) ? String.format("https://api.%s", host)
+                                       : String.format("https://%s/api/v3", host);
```
Comment (vikyathharekal) [inline idp-service/src/main/java/io/harness/idp/gitintegration/processor/impl/GithubConnectorProcessor.java:156]: We might have to make this configurable in the future. Looks good for now!

File: idp-service/src/main/java/io/harness/idp/gitintegration/service/GitIntegrationServiceImpl.java [hunk 4]
```diff
@@ -55,16 +51,11 @@ public class GitIntegrationServiceImpl implements GitIntegrationService {
   DelegateSelectorsCache delegateSelectorsCache;
   ProxyEnvVariableUtils proxyEnvVariableUtils;
 
-  private static final String TARGET_TO_REPLACE_IN_CONFIG = "HOST_VALUE";
-
   private static final String SUFFIX_FOR_GITHUB_APP_CONNECTOR = "_App";
```
Comment (vikyathharekal) [inline idp-service/src/main/java/io/harness/idp/gitintegration/service/GitIntegrationServiceImpl.java:54]: These constants might not be needed here anymore

**Comment by vikyathharekal** [comment]
Looks like some bazel deps have to be fixed in tests as well.

File: idp-service/src/main/java/io/harness/idp/app/BUILD.bazel [hunk 5]
```diff
@@ -39,7 +39,13 @@ java_binary(
         "//idp-service/src/main/java/io/harness/idp/allowlist:module",
         "//idp-service/src/main/java/io/harness/idp/common/delegateselectors/cache:module",
         "//idp-service/src/main/java/io/harness/idp/common/delegateselectors/cache/memory:module",
-        "//idp-service/src/main/java/io/harness/idp/configmanager:module",
+        "//idp-service/src/main/java/io/harness/idp/configmanager/beans/entity:module",
+        "//idp-service/src/main/java/io/harness/idp/configmanager/jobs:module",
+        "//idp-service/src/main/java/io/harness/idp/configmanager/mappers:module",
```
[the remark this answers is not in the document]
Comment (devesh-harness) [reply idp-service/src/main/java/io/harness/idp/app/BUILD.bazel]: done

File: idp-service/src/main/java/io/harness/idp/configmanager/beans/entity/BUILD.bazel [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 Harness Inc. All rights reserved.
+# Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+# that can be found in the licenses directory at the root of this repository, also available at
+# https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+
+load("//:tools/bazel/harness.bzl", "java_library")
+load("//:tools/bazel/GenTestRules.bzl", "run_tests")
```
[the remark this answers is not in the document]
Comment (devesh-harness) [reply idp-service/src/main/java/io/harness/idp/configmanager/beans/entity/BUILD.bazel]: done

File: idp-service/src/main/java/io/harness/idp/configmanager/mappers/BUILD.bazel [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2023 Harness Inc. All rights reserved.
+# Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+# that can be found in the licenses directory at the root of this repository, also available at
+# https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+
+load("//:tools/bazel/harness.bzl", "java_library")
+load("//:tools/bazel/GenTestRules.bzl", "run_tests")
```
[the remark this answers is not in the document]
Comment (devesh-harness) [reply idp-service/src/main/java/io/harness/idp/configmanager/mappers/BUILD.bazel]: done

File: idp-service/src/main/java/io/harness/idp/configmanager/repositories/BUILD.bazel [hunk 8]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2023 Harness Inc. All rights reserved.
+# Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+# that can be found in the licenses directory at the root of this repository, also available at
+# https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+
+load("//:tools/bazel/harness.bzl", "java_library")
+load("//:tools/bazel/GenTestRules.bzl", "run_tests")
```
[the remark this answers is not in the document]
Comment (devesh-harness) [reply idp-service/src/main/java/io/harness/idp/configmanager/repositories/BUILD.bazel]: done

File: idp-service/src/main/java/io/harness/idp/configmanager/repositories/BUILD.bazel [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+load("//:tools/bazel/macros.bzl", "run_analysis")
+
+java_library(
+    name = "module",
+    srcs = glob(["**/*.java"]),
+    resources = [],
+    visibility = ["//idp-service:__subpackages__"],
+    deps = [
+        "//:lombok",
+        "//idp-service/contracts/openapi/v1:idp-service-server-spec-module",
```
[the remark this answers is not in the document]
Comment (devesh-harness) [reply idp-service/src/main/java/io/harness/idp/configmanager/repositories/BUILD.bazel:15]: ok

File: idp-service/src/main/java/io/harness/idp/proxy/delegate/BUILD.bazel (hunk 1 above)
Comment (devesh-harness) [reply idp-service/src/main/java/io/harness/idp/proxy/delegate/BUILD.bazel:19]: ok

File: idp-service/src/main/java/io/harness/idp/proxy/envvariable/BUILD.bazel (hunk 2 above)
Comment (devesh-harness) [reply idp-service/src/main/java/io/harness/idp/proxy/envvariable/BUILD.bazel]: done

File: idp-service/src/main/java/io/harness/idp/gitintegration/processor/impl/GithubConnectorProcessor.java (hunk 3 above)
Comment (devesh-harness) [reply idp-service/src/main/java/io/harness/idp/gitintegration/processor/impl/GithubConnectorProcessor.java:156]: ok

File: idp-service/src/main/java/io/harness/idp/gitintegration/service/GitIntegrationServiceImpl.java (hunk 4 above)
Comment (devesh-harness) [reply idp-service/src/main/java/io/harness/idp/gitintegration/service/GitIntegrationServiceImpl.java]: done

**Comment by devesh-harness** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger uts

File: idp-service/src/main/resources/configs/integrations/gitlab.yaml [hunk 10]
```diff
@@ -1,4 +1,5 @@
 integrations:
   gitlab:
     - host: HOST_VALUE
+      apiBaseUrl: https://HOST_VALUE/api/v4
```
Comment (vigneswara-propelo) [inline idp-service/src/main/resources/configs/integrations/gitlab.yaml:4]: we don't need to mention `apiBaseUrl` for Bitbucket server?
Ref: https://backstage.io/docs/integrations/bitbucketServer/locations

Comment (devesh-harness) [reply idp-service/src/main/resources/configs/integrations/gitlab.yaml:4]: it is present already for bitbucket server

**Review by vigneswara-propelo — APPROVED** [review]

**Review by vikyathharekal — APPROVED** [review]

**Comment by vikyathharekal** [comment]
trigger review

**Pull request merged**